### ISSUE TITLE
Redis integration fails to run cli (Illegal option -o pipefail)

### ISSUE THREAD

**Issue by icanhazstring** [drud/ddev-contrib#80]
Hi, I used the redis configuration as provided.
But running the command `ddev redis-cli` results in the following error:

```
$ ddev redis-cli
sh: 1: set: Illegal option -o pipefail
Failed to run redis-cli : exit status 2
```

Ddev version
```
$ ddev --version
ddev version v1.15.0
```

By searching for it I only found out, that this is an error if you are running the command using `/bin/sh` which links to `/bin/dash` which does not have the option `-o`.

The `docker-entrypoint.sh` from redis is pretty solid and has a shebang for `/bin/bash` in it, so this should be fine.
Running the docker container directly works:

```
$ docker exec -i 2e7f7ec00f28 redis-server --version
Redis server v=6.0.5 sha=00000000:0 malloc=jemalloc-5.1.0 bits=64 build=db63ea56716d515f
```

Is there any special execution happening behind the scenes using ddev which runs the command using `/bin/sh` or something?

**Comment by rfay** [comment]
ddev v1.15 is the first version that checks for errors in script when they run.

**Comment by icanhazstring** [comment]
Well the custom command itself is nothing fancy. If I do a `docker exec` on the container itself with just the same command as in https://github.com/drud/ddev-contrib/blob/master/docker-compose-services/redis/commands/redis/redis-cli. Everything works.

**Comment by rfay** [comment]
The problem here is that the default shell in the redis container is sh, not bash. 

This is actually a bug in ddev. 

https://github.com/drud/ddev/blob/9bb78d5ba819f3f9a1907995eb9e07de79d67417/pkg/ddevapp/ddevapp.go#L1068

What's going on is that ddev doesn't assume that bash is the default shell (or even available) in non-ddev services. So here it's assuming that it should use `sh` to execute the script, and bourne shell doesn't allow pipefail.

**Comment by rfay** [comment]
If all goes well this will be fixed in ddev v1.15.1, see https://github.com/drud/ddev/pull/2395

**Comment by rfay** [comment]
This will be fixed by deev v1.15.1 tomorrow. Thanks for reporting it, it was important!

**Comment by icanhazstring** [comment]
Thanks it works. Much appreciate it 🙏
